Jonathen Cheng Yuzhe ip

jonaturn opened the pull request:
BotJohn frees your mind of having to remember your todos. It's
- text-based
- easy to pick up
- ~~FAST~~ SUPER FAST

All you need to do is,

1. download it from [here](https://github.com/jonaturn/ip/tree/master)
2. double click it
3. add your task
4. let John manage your deadlines 😄 

At no cost:

Features: 

- [x] Manage Tasks
- [ ] Manage Deadlines (coming soon frfr)
- [ ] Reminders (coming soon frfr)

If you are a Java programmer, you can add your own features too. Start with the `main` method and branch out as you go.

```
public static void main(String[] args) throws Exception {
        // init
        new John().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jonaturn committed:
level-0

jonaturn committed:
level 1 echo

jonaturn committed:
level-2 add list

jonaturn committed:
level-3 mark as done

jonaturn committed:
level 4 added todos, events, deadlines

jonaturn committed:
level 5 added exception handling

jonaturn committed:
todo: improve the string input handling

jonaturn committed:
A-enums, input stringhandling using enums

jonaturn committed:
git branches

jonaturn committed:
git merge

jonaturn committed:
merge conflict fix

jonaturn committed:
merge example again

jonaturn committed:
added filewrite

jonaturn committed:
Merge branch 'master' of https://github.com/jonaturn/ip

jonaturn committed:
added localdatetime

jonaturn committed:
add OOP and Packages

jonaturn committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2425S2/ip into add-gradle-support

jonaturn committed:
add packages

jonaturn committed:
add unit tests

jonaturn committed:
add unit tests

jonaturn committed:
A-JavaDoc

jonaturn committed:
add find

jonaturn committed:
A-Jar

jonaturn committed:
add jar

jonaturn committed:
A-CheckStyle

jonaturn committed:
add gui

vrisdng commented on src/main/java/John.java:
```diff
@@ -0,0 +1,53 @@
+     * @throws IOException
+     */
+    public void run() throws IOException {
+        Scanner sc = new Scanner(System.in);
```
Since we are not taking user's input anymore, but loading it from a file and saving it to a file, where did you put this in your code?

vrisdng commented on src/main/java/parser/InputType.java:
```diff
@@ -0,0 +1,43 @@
+/**
+ * Enums for commands
+ */
+public enum InputType {
```
Nice way to categorizes the input types here, but maybe you can extract it a little bit more? e.g: INVALID is not really an input type but should be an exception

vrisdng commented on src/main/java/tasks/Todo.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Class that handles todo tasks
+ */
+public class Todo extends Task {
```
It feels quite weird to me that there are no methods in the subclasses of task, perhaps you omitted it? Maybe it is your program's choice of design, but I would recommend you to have methods that are specific to this class, e.g: An override toString method to describe Todo tasks.

vrisdng reviewed (commented):
Overall, you had a clear structure of code. There are some parts that have rooms for improvements. On additional note, your repository's name should be in the form of [your name] iP, else they might not detect yours correctly?

Charlesl12 commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+    @Override
+    public String name() {
+        return super.name() + " (by: " + by.format(DateTimeFormatter.ofPattern("dd MMM yyyy h:mma")) + ")";
+    }
```
I think the method name could be improved here! Perhaps getName() or getDescription?

Charlesl12 reviewed (commented):
Great JavaDoc with complying conventions. Although some methods could follow the get...() naming conventions. LGTM.

jonaturn committed:
finish backlog

jonaturn committed:
fixed GUI

jonaturn committed:
added assert

jonaturn committed:
Fixed:\nNot loading full list of tasks from file\nFile type was not initialised properly on load\n\nCode refactoring:\nCreated new Exception classes to abstract away code from Parser.java\nCreated new methods in UI package to handle creating string responses for John

jonaturn committed:
Fixed:

Not loading full list of tasks from file

File type was not initialised properly on load

Code refactoring:

Created new Exception classes to abstract away code from Parser.java\nCreated new methods in UI package to handle creating string responses for John

jonaturn committed:
Added Tagging:\n\n[tag] [index(int)] [tagDescription(String)]\n'tag' as keyword\nInteger for task index \nDescription of tag

jonaturn committed:
Added Tagging:\n\n[tag] [index(int)] [tagDescription(String)]\n'tag' as keyword\nInteger for task index \nDescription of tag

jonaturn committed:
Fixed:

Not loading full list of tasks from file

File type was not initialised properly on load

Code refactoring:

Created new Exception classes to abstract away code from Parser.java\nCreated new methods in UI package to handle creating string responses for John

jonaturn committed:
fix branch lvl10

jonaturn committed:
level-9

jonaturn committed:
level-9

jonaturn committed:
A-JavaDoc

jonaturn committed:
A-JavaDoc

jonaturn committed:
branch-A-CodingStandard

jonaturn committed:
branch-A-CodingStandard

jonaturn committed:
BCD-Extension

jonaturn committed:
Full \n test

jonaturn committed:
Fixed:

Not loading full list of tasks from file

File type was not initialised properly on load

Code refactoring:

Created new Exception classes to abstract away code from Parser.java\nCreated new methods in UI package to handle creating string responses for John

jonaturn committed:
Revert "Full \n test"

This reverts commit 75b8ed66da4d13722d6b12d204f9309db49fffad.

jonaturn committed:
Merge branch 'master' of https://github.com/jonaturn/ip

jonaturn committed:
fixed master branch

Messed up some rebase

push from local to restore

jonaturn committed:
A
Merge branch 'master' of https://github.com/jonaturn/ip
Did not include full commit messages.
Attempted to rebase and ammend commit messages

certain commmits got messed up

pulling from local to push and restore

jonaturn committed:
Repo did not include FullCommit Message

To practise and familiarise with full commit messages

Created a new branch to learn full commit messages

jonaturn committed:
Repo did not include FullCommit Message

To practise and familiarise with full commit messages

Created a new branch to learn full commit messages

jonaturn committed:
fix branch name

jonaturn committed:
Add resize window

jonaturn committed:
Change GUI theme

jonaturn committed:
Add wrong input message

jonaturn committed:
Remove redundant button

jonaturn committed:
Fix dialog-box

jonaturn committed:
Add background

jonaturn committed:
Add background

jonaturn committed:
Add ui

jonaturn committed:
Finalise project

jonaturn committed:
Fix dependencies

jonaturn committed:
Add UserGuide

jonaturn committed:
Fix bugs

jonaturn committed:
Fix deadline bug

jonaturn committed:
Finalise submission

jonaturn committed:
Add graceful exception catch

jonaturn committed:
Fix Ui.png

jonaturn committed:
Fix storage

jonaturn committed:
Fix readme
